### DISCUSSION TITLE
multiapp transfer vpp of type ElementValueSampler

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#22482] [Q&A General] [answered]
Hi 

I need to transfer vpp of type ` ElementValueSampler` from the sub-app to the main-app reporter.
The block that describes the transfer in the main app is as follows:

```
  [fromadjoint]
    type = MultiAppReporterTransfer
    from_multi_app = adjoint
    from_reporters = 'gradvec/gradvec'
    to_reporters = 'OptimizationReporter/adjoint'
  []
```
and the vpp in the sub-app is as follows:

```
[VectorPostprocessors]
  [gradvec]
    type = ElementValueSampler
    variable = 'integral_gradient'
    sort_by = id
    execute_on = 'final'
  []
[]
```

This results in the following error 
```
"*** ERROR ***
The following error occurred in the object "fromadjoint", of type "MultiAppReporterTransfer".
Reporter with the name "gradvec/gradvec" within app "adjoint0" was not found."
```

@lynnmunday any thoughts from your side?

**Comment by GiudGiud** [comment, marked as answer]
The name must not be gradvec. 
Look for the declareVec name in the source code to get the right name used in the vector

**Reply by aaelmeli** [reply]
> The name must not be gradvec. Look for the declareVec name in the source code to get the right name used in the vector

It seems that `ElementValueSampler`  does not have a variable that holds a name for the vector.

**Reply by GiudGiud** [reply]
ElementValueSampler just uses the names of the variables for the vector

**Reply by aaelmeli** [reply]
> ElementValueSampler just uses the names of the variables for the vector

Thank you, that helps. it should be modified to  `from_reporters = 'gradvec/integral_gradient'`. Thanks again.

**Reply by lynnmunday** [reply]
@aaelmeli I just saw this, I hope you figured it out.  I ran into this same problem.  We automatically create a reporter that appends "grad_" to the parameter_names that are given in the OptimizationReporter block.  
See this example in moose/modules/optimization/test/tests/optimizationreporter/bc_load_linearFunction/main.i

Note teh parameter_names 'left right', these will be used to create two reporters that we need to transfer the gradient data into called grad_left and grad_left.

``` [OptimizationReporter]
  type = OptimizationReporter
  parameter_names = 'left right'
  num_values = '2 1'
  measurement_file = 'measurementData.csv'
  file_xcoord = 'coordx'
  file_ycoord ='y'
  file_zcoord = 'z'
  file_value = 'measured_value'
  # contents of measurementData.csv
  # measurement_points = '0.2 0.2 0
  #                       0.8 0.6 0
  #                       0.2 1.4 0
  #                       0.8 1.8 0'
  # measurement_values = '207 204 185 125'
  outputs=none
[]
```
and then to get this transfer to work, we append grad_ to the two parameter names for the to_reporters shown below by grad_left and grad_right

```
  [fromadjoint]
    type = MultiAppReporterTransfer
    from_multi_app = adjoint
    from_reporters = 'grad_bc_left/inner_product
                      grad_bc_right/inner_product'
    to_reporters = 'OptimizationReporter/grad_left
                    OptimizationReporter/grad_right'
  []
```

**Reply by lynnmunday** [reply]
A good way to check what reporters are getting created in your problem is to use this debug block:
```
[Debug]
  show_reporters = true
[]
```

**Reply by aaelmeli** [reply]
> @aaelmeli I just saw this, I hope you figured it out. I ran into this same problem. We automatically create a reporter that appends "grad_" to the parameter_names that are given in the OptimizationReporter block. See this example in moose/modules/optimization/test/tests/optimizationreporter/bc_load_linearFunction/main.i
> 
> Note teh parameter_names 'left right', these will be used to create two reporters that we need to transfer the gradient data into called grad_left and grad_left.
> 
> ```
>   type = OptimizationReporter
>   parameter_names = 'left right'
>   num_values = '2 1'
>   measurement_file = 'measurementData.csv'
>   file_xcoord = 'coordx'
>   file_ycoord ='y'
>   file_zcoord = 'z'
>   file_value = 'measured_value'
>   # contents of measurementData.csv
>   # measurement_points = '0.2 0.2 0
>   #                       0.8 0.6 0
>   #                       0.2 1.4 0
>   #                       0.8 1.8 0'
>   # measurement_values = '207 204 185 125'
>   outputs=none
> []
> ```
> 
> and then to get this transfer to work, we append grad_ to the two parameter names for the to_reporters shown below by grad_left and grad_right
> 
> ```
>   [fromadjoint]
>     type = MultiAppReporterTransfer
>     from_multi_app = adjoint
>     from_reporters = 'grad_bc_left/inner_product
>                       grad_bc_right/inner_product'
>     to_reporters = 'OptimizationReporter/grad_left
>                     OptimizationReporter/grad_right'
>   []
> ```

Thanks, @lynnmunday, I could fix this problem at that time. I think, by that time, that appended (grad_) was not there OR my repository was not updated.
